spacesvm/src/api/client.rs

```diff
@@ -1,48 +1,80 @@
 use std::{
     fs::File,
-    io::{Result, Write},
+    io::{Error, ErrorKind, Result, Write},
     path::Path,
+    sync::Arc,
 };
 
 use crate::{
     api::{
         DecodeTxArgs, DecodeTxResponse, IssueTxArgs, IssueTxResponse, PingResponse, ResolveArgs,
         ResolveResponse,
     },
-    chain::tx::{tx::TransactionType, unsigned::TransactionData},
+    chain::tx::{
+        decoder::{self, TypedData},
+        tx::TransactionType,
+        unsigned::TransactionData,
+    },
+};
+use avalanche_types::key::{
+    self,
+    secp256k1::{private_key::Key, signature::Sig},
 };
-use avalanche_types::key;
 use http::{Method, Request};
 use hyper::{body, client::HttpConnector, Body, Client as HyperClient};
-use jsonrpc_core::{Call, Id, MethodCall, Params, Version};
+use jsonrpc_core::{Call, Id, MethodCall, Params, Value, Version};
 use serde::de;
 
 pub use http::Uri;
+use tokio::sync::RwLock;
 
-/// HTTP client for interacting with the API, assumes single threaded use.
+/// HTTP client for interacting with the API.
```

Comment:
nit: The `Arc` makes it obvious but looks like the Client supports multi-threaded use now. Worth adding a comment?